dashboard/streamlit.py

```diff
@@ -1,6 +1,57 @@
 import pandas as pd
+import os
 import streamlit as st
 import altair as alt
+import pyodbc
+import logging
+from dotenv import load_dotenv
+
+load_dotenv('.env')
+DB_HOST = os.getenv('DB_HOST')
+DB_PORT = os.getenv('DB_PORT')
+DB_USER = os.getenv('DB_USER')
+DB_PASSWORD = os.getenv('DB_PASSWORD')
+DB_NAME = os.getenv('DB_NAME')
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class DataLoader:
+
+    @staticmethod
+    def connect_to_db(db_host: str = DB_HOST, db_port: str = DB_PORT,
+                      db_user: str = DB_USER, db_password: str = DB_PASSWORD,
+                      db_name: str = DB_NAME):
+        """Connects to the Microsoft SQL Server database"""
+        try:
+            connection_string = f"DRIVER={{ODBC Driver 17 for SQL Server}};SERVER={
+                db_host},{db_port};DATABASE={db_name};UID={db_user};PWD={db_password}"
+            connection = pyodbc.connect(connection_string)
+            logger.info("Database connection established successfully.")
+            return connection
+        except Exception as e:
+            logger.error("Failed to connect to the database: %s", e)
+            raise
+
+    @staticmethod
+    def fetch_data_from_tables(connection):
+        """Fetches data from two tables and returns as dataframes"""
+        try:
+            query1 = "SELECT * FROM beta.reading"
+            query2 = "SELECT * FROM beta.plant"
+            query3 = "SELECT * FROM beta.region"
+            query4 = "SELECT * FROM beta.botanist"
+
+            reading_df = pd.read_sql(query1, connection)
+            plant_df = pd.read_sql(query2, connection)
+            region_df = pd.read_sql(query3, connection)
+            botanist_df = pd.read_sql(query4, connection)
+
+            logger.info("Data fetched successfully from tables.")
+            return reading_df, plant_df, region_df, botanist_df
+        except Exception as e:
+            logger.error("Failed to fetch data from tables: %s", e)
+            raise
 
 
 class dashboard:
```

Comment:
There's very little for me to comment on here, the classes all look sensible and the final dashboard looks good.